Deployment

yfg-dev opened the pull request:
- set HOST_ADDRESS e.g. localhost in Peerprep .env
- use the HOST_ADDRESS to set matching service, user service and question service end point in nuxt.config.ts
- Add unit testing code for matching service

jamieeeleow committed:
Implemented CodeMirror for code editor

jamieeeleow committed:
Create new Firestore Document based on Session ID

jamieeeleow committed:
Fix race condition between Firestore updates and CodeMirror changes

jamieeeleow committed:
Incorporate Liveblocks for real-time synchronisation of CodeMirror

yfg-dev committed:
Add deployment configs

yfg-dev committed:
Add deployment configs

yfg-dev committed:
Add unit testing for matching service

yfg-dev committed:
Fetch attempt histories with environment vairables

- set the HOST_ADDRESS variable in .env
- use the runtimeConfig to define the endpoint to matching service, question service and use service

yfg-dev committed:
Merge branch 'matching-service' into deployment

jamieeeleow committed:
Implemented syntax highlighting for javascript

jamieeeleow committed:
Fetch question and include termination button for collaboration

yfg-dev committed:
Merge branch 'matching-service' into deployment

Kidsnd274 commented on frontend/Peerprep/README.md:
```diff
@@ -12,9 +12,10 @@ npm install
 - Set up environment varaibles in a `.env` file
   - `FIREBASE_API_KEY` which you can get from the firebase console.
-
+  - `HOST_ADDRESS` = localhost or the static ip.
```
Does this mean that the microservices are exposed to the internet?

Kidsnd274 commented on frontend/Peerprep/README.md:
```diff
@@ -12,9 +12,10 @@ npm install
 - Set up environment varaibles in a `.env` file
   - `FIREBASE_API_KEY` which you can get from the firebase console.
-
+  - `HOST_ADDRESS` = localhost or the static ip.
```
Does this mean that the microservices are exposed to the Internet?

yfg-dev replied:
yea, shld be able to be access with port number

Kidsnd274 replied:
Is there a way to keep it hidden from the internet, so that only the microservices can access them? Because the backend APIs are not supposed to be internet facing.

yfg-dev replied:
use server actions for frontend to call backend services except for websocket. therefore, no need to expose the backend apis except for websocker connection

Huggenguggen committed:
Add unit tests for question_service

Add unit tests using pytest to question service
Tests both kafka-consumer and routes.py
Add github workflow as well

Huggenguggen committed:
change pytest.yaml

Add a on.pull_request.branches for it to be triggered on PR

Huggenguggen committed:
change pytest.yaml

Remove the on_workflow rule as it was skipping the action

Huggenguggen committed:
change pytest.yaml

remove the if as it was skipping the action

Huggenguggen committed:
edit pytest.yaml

setting up environment variables was erroneous

Huggenguggen committed:
edit pytest.yaml

Change the envrionment variables again as well as the way it is run

Huggenguggen committed:
edit pytest.yaml

add pwd for debugging

Huggenguggen committed:
edit pytest.yaml

Huggenguggen committed:
edit pytest.yaml

Try using base64 for secret

Huggenguggen committed:
edit pytest.yaml

Huggenguggen committed:
edit pytest.yaml

yfg-dev committed:
Hide navbar in collaboration page and disable ridrect when in collaboration session

yfg-dev committed:
use server actions for api calls

yfg-dev committed:
use server actions for api calls

Kidsnd274 committed:
Remove extra console log statement

HT-T committed:
implements termination

on clicking the termination button, user is moved back to home page

yfg-dev committed:
Merge branch 'main' of https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g29 into deployment

Huggenguggen reviewed (approved):
LGTM
